api/src/legacy_soap_api/applicants/services/get_opportunity_list_response.py

```diff
@@ -38,6 +49,43 @@ def get_competitions_from_opportunity_list_request(
     )
 
 
+def get_tinstructions_map(db_session: db.Session, competitions: list) -> dict:
+    """Get instructions extension for competitions
+
+    This method returns a map of extension string from the staging.tinstructions table
+    that is needed to build the instructions_url property of the competition for the
+    GetOpportunityListResponse.
+
+    This will return a map of Competition.legacy_competition_id: staging.tinstructions.extension.
+    """
+    legacy_competition_ids = [
+        c.legacy_competition_id for c in competitions if c.legacy_competition_id is not None
+    ]
+    tinstructions = (
+        db_session.execute(
+            select(Tinstructions.extension, Tinstructions.comp_id).where(
+                Tinstructions.comp_id.in_(legacy_competition_ids)
+            )
+        )
+        .mappings()
+        .all()
+    )
+    if not tinstructions:
+        return {}
+    return {tinstruction.comp_id: tinstruction.extension for tinstruction in tinstructions}
```

Comment:
I believe if the list is empty, the generator should produce the same result.
```suggestion
    return {tinstruction.comp_id: tinstruction.extension for tinstruction in tinstructions}
```